src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
+            values = new Node[values.length * 2];
+            for (int i = 0; i < oldBucketStock.length; i++) {
+                while (oldBucketStock[i] != null) {
+                    K k = oldBucketStock[i].key;
+                    V v = oldBucketStock[i].value;
+                    int index = getIndexFromHash(k);
+                    if (values[index] == null) {
+                        values[index] = new Node<>(k, v);
+                    } else {
+                        Node<K, V> lastBucketNode = values[index];
+                        while (lastBucketNode.nextNode != null) {
+                            lastBucketNode = lastBucketNode.nextNode;
+                        }
+                        lastBucketNode.nextNode = new Node<>(k, v);
+                    }
+                    oldBucketStock[i] = oldBucketStock[i].nextNode;
+                }
+            }
+            this.threshold = (int) (values.length * LOAD_FACTOR);
+        }
+        int index = getIndexFromHash(key);
+        if (values[index] == null) {
+            values[index] = new Node<>(key, value);
+        } else {
+            Node<K, V> lastBucketNode = values[index];
+            if (lastBucketNode.key == null && key == null) {
+                lastBucketNode.value = value;
+                return;
+            } else if (key != null && key.equals(lastBucketNode.key)) {
+                lastBucketNode.value = value;
+                return;
+            }
+            while (lastBucketNode.nextNode != null) {
+                if (Objects.equals(lastBucketNode.key, key)) {
+                    lastBucketNode.value = value;
+                    return;
+                }
+                lastBucketNode = lastBucketNode.nextNode;
+            }
+            lastBucketNode.nextNode = new Node<>(key, value);
+        }
+        size = size + 1;
     }
 
+
+
     @Override
     public V getValue(K key) {
+        int index = getIndexFromHash(key);
+        Node<K, V> returnNode = values[index];
+        while (returnNode != null) {
+            /* if (key == returnNode.key) {
+                return returnNode.value;
+            } */
+            if (key == null && returnNode.key == null) {
+                return returnNode.value;
+            }
+            if (key != null && key.equals(returnNode.key)) {
+                return returnNode.value;
+            }
+            returnNode = returnNode.nextNode;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(int hashCode, int primaryNumber) {
+        hashCode ^= (hashCode >>> 20) ^ (hashCode >>> 12);
+        return hashCode ^ (hashCode >>> 7) ^ (hashCode >>> 4);
+
+    }
+
+    private int generateRandomPrimaryNumber() {
+        int n = new Random().nextInt(100);
+        for (int i = 2; i < n; i++)
+            if (n % i == 0)
+                return n;
+        return 1;
```

Comment:
1 is not a primary number - if we want to keep this logic change to return 2 as it is first primary number